Dockerfile

```diff
@@ -0,0 +1,28 @@
+FROM python:3.8-slim-buster AS base
+LABEL key="Jeremy Brex todo_app"
+
+# Add Curl as not part of slim Buster
+RUN apt-get update -y && apt-get install curl -y
+
+# Install Poetry
+
+ENV POETRY_HOME=/poetry
+ENV PATH=${POETRY_HOME}/bin:${PATH}
+
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+
+# Set the working directory.
+WORKDIR /app
+COPY todo_app /app/todo_app
+COPY pyproject.toml /app
+COPY poetry.toml /app
```

Comment:
You should also copy the .lock file and not delete it!
It keeps versions of dependencies fixed precisely unless you choose to update them. The pyproject.toml file has more lax requirements, e.g. `^` means it the last part of the version is *at least* the number given.